code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -0,0 +1,194 @@
+	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	AddComponent(/datum/component/clickbox, x_offset = -2, y_offset = -2)
+	AddComponent(/datum/component/swarming)
+	AddComponent(/datum/component/squashable, squash_chance = 35, squash_damage = 1)
+	AddComponent(/datum/component/tameable, food_types = list(/obj/item/food), tame_chance = 30, copy_faction = FALSE, bonus_tame_chance = 20)
+
+/mob/living/simple_animal/hostile/cockroach/GiveTarget(new_target)
+	add_target(new_target)
+	LosePatience()
+	if(target != null)
+		if(istype(target, /obj/item/food))
+			GainPatience()
+			Aggro()
+			retreat_distance = 0
+			minimum_distance = 0
+			return TRUE
+		else if(isliving(target))
+			GainPatience()
+			Aggro()
+			if(explosive)
+				retreat_distance = 0
+				minimum_distance = 1
+				return TRUE
+			visible_message(span_danger("[name] scuttles away!"))
+			retreat_distance = 10
+			minimum_distance = 10
+			return TRUE
+
+/mob/living/simple_animal/hostile/cockroach/AttackingTarget()
+	if(istype(target, /obj/item))
+		consume(target)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/cockroach/proc/consume(atom/movable/fooditem)
+	playsound(src, 'sound/items/eatfood.ogg', 100, FALSE)
+	visible_message(span_danger("[name] devours [fooditem]!"))
+	qdel(fooditem)
+	if(sterile)
+		return
+	var/cap = CONFIG_GET(number/roachcap)
+	if(LAZYLEN(SSmobs.roaches) >= cap)
+		return
+	new /mob/living/simple_animal/hostile/cockroach(get_turf(src))
+
+/mob/living/simple_animal/hostile/cockroach/attackby(obj/item/I, mob/living/user)
+	var/hot_thing = I.get_temperature()
+	if(hot_thing) //FIRE IS THE BEST WAY TO KILL A SPIDER
+		user.visible_message(span_warning("[user] lights [src] on fire!"), span_warning("You light [src] on fire!"))
+		burner = TRUE
+		update_icon()
+		addtimer(CALLBACK(src, /mob/living/.proc/death), 10 SECONDS) //just in case the fire doesn't kill it.
+		search_objects = 0 //run away from people
+		qdel(GetComponent(/datum/component/squashable)) //you aren't getting out of this mess that easy
+		return
+	if(istype(I, /obj/item/grenade))
+		attach_nade(I, user)
+		update_icon()
+		return
+	if(istype(I, /obj/item/wirecutters))
+		if(explosive)
+			if(prob(explode_chance + 20))
+				user.visible_message(span_warning("[src] fidgets as [user] tries to disarm it! OH SHIT-"), span_warning("[src] fidgets as you try to disarm it! OH SHIT-"))
+				death() //cockroach death triggers the bomb
+			to_chat(user, span_notice("You disarm [src]."))
+			var/turf/roach_turf = get_turf(src)
+			explosive.forceMove(roach_turf)
+			explosive = null
+			return
+		update_icon()
+	return ..()
+
+/mob/living/simple_animal/hostile/cockroach/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	playsound(src, 'sound/creatures/roach_scuttle.ogg', 50, FALSE)
+	if(burner)
+		//currently on fire.
+		var/turf/fire_turf = get_turf(src)
+		addtimer(CALLBACK(src, .proc/setFire, fire_turf), 0.6 SECONDS)
```

Comment:
why is this even a timer

Reply:
The delayed fire trail looks funnier. I might just change this to light the turf they left on fire instead, though

Reply:
please do. timers are expensive